src/ducks/triggers/index.js

```diff
@@ -46,14 +48,42 @@ export const deleteTrigger = trigger =>
 export const launchTrigger = trigger => fromCozyClient.launchTrigger(trigger)
 
 // Helpers
+export const buildTriggerFrequencyOptions = (konnector, options) => {
+  const { frequency } = konnector
+  const { day, hours, minutes } = options
+
+  const frequencyOptions = {
+    frequency:
+      frequency && VALID_FREQUENCIES.includes(frequency) ? frequency : 'weekly'
+  }
+
+  if (frequencyOptions.frequency === 'daily') {
+    return {
+      ...frequencyOptions,
+      hours,
+      minutes
+    }
+  }
+
+  // weekly case by default
+  return {
+    ...frequencyOptions,
+    day,
+    hours,
+    minutes
+  }
```

Comment:
Do you see a problem doing it like so ? I find it clearer : 

```jsx
  const parseFrequency = freq => freq && VALID_FREQUENCIES.includes(freq) ? freq : 'weekly'
  const frequencyOptions = {
    hours,
    minutes,
    frequency: parseFrequency(frequency)
  }
 
  if (freq === 'weekly') {
    frequencyOptions.day = day
  }

  return frequencyOptions
```

Reply:
Ok for me except the mutation of `frequencyOptions` 👍 

Please have a look at the last commit.